core/src/main/java/org/apache/iceberg/GenericManifestEntry.java

```diff
@@ -156,6 +183,9 @@ public void put(int i, Object v) {
         this.sequenceNumber = null;
         return;
       case 3:
+        this.fileSequenceNumber = (Long) v;
```

Comment:
Okay, it seems safe as we use a projection there. That being said, I will need to adapt it in a follow-up to persist the file sequence number as well.

Reply:
Should we do it in this PR rather than in a follow up? That seems like an important part and we don't want to release this without it.

Reply:
I agree that part must be done before releasing `fileSequenceNumber`. However, I'd do it in a separate PR immediately after this one as `fileSequenceNumber` is optional and the current logic does not cause correctness issues. It would be just easier to separate these two.